core/domain/exp_domain.py

```diff
@@ -533,35 +532,30 @@ def validate(self, obj_type, exp_param_specs_dict):
                 'There must be at least one rule for each answer group.'
                 % self.rule_specs)
 
-        all_rule_classes = rule_domain.get_rules_for_obj_type(obj_type)
-        seen_fuzzy_rule = False
+        seen_classifier = False
```

Comment:
Perhaps we should instead refer to ``classifier_rule`` here and in reader.py, to be consistent with the name of the method. I think that's a clearer name than just ``classifier``, since the latter is ambiguous with statistical classifier over a rule which contains training data for a statistical classifier.